### PR TITLE
Source as s3 bucket

### PR DESCRIPTION
**Pull request by curzio-della-santa**

### CODE REVIEW

**Comment by curzio-della-santa** [comment]
2017-04-22 09_42_46-blitline examples web01 detailnet me_job-submit php


2017-04-22 09_42_57-blitline examples web01 detailnet me_job-info php_job_id 9srokkvodcoobbxxd8icwqg


2017-04-22 09_42_22-blitline _ saas services landing

File: examples/config.php.dist [hunk 1]
```diff
@@ -3,5 +3,8 @@
 return array(
     'application_id' => 'your-blitline-application-id',
     's3bucket' => 'your-bucket-name',
-    's3path' => 'test-files', // Or any other path under your bucket
+    's3prefix' => 'test-files', // Or any other path under your bucket
+    //'s3file' => 'Desert.jpg', // Or any other file under your bucket/prefix
+    's3file' => 'Tulips-private.jpg',
+    //'s3file' => 'mountains.tif',
```
Comment (ivan-wolf) [inline examples/config.php.dist]: Does it make sense to configure a file here?

And please clean it up.

File: examples/job-submit.php [hunk 2]
```diff
@@ -31,14 +31,19 @@
...
 );
 
 $job = $jobBuilder->createJob()
-    ->setSourceUrl($imageUrl)
-    ->addFunction(
+    ->setSource(
+        array(
+            'name' => 's3',
+            'bucket' => $getConfig('s3bucket'),
+            'key' => $getConfig('s3prefix') . '/' . $getConfig('s3file'),
```
Comment (ivan-wolf) [inline examples/job-submit.php]: Read file/path from URL param?

File: examples/job-submit.php [hunk 3]
```diff
@@ -53,7 +58,7 @@
                     'image_identifier' => $imageName,
                     's3_destination' => array(
 //                        'bucket' => $getConfig('s3bucket'),
-                        'key' => $getConfig('s3path') . '/' . $imageName . '-' . $imageSize . '_blitline.jpg',
+                        'key' => $getConfig('s3prefix') . '/' . $imageName . '-' . $imageSize . '_blitline.jpg',
```
Comment (ivan-wolf) [inline examples/job-submit.php:58]: Extract prefix from given file/path.

File: src/Detail/Blitline/Client/ServiceDescription/Blitline.php [hunk 4]
```diff
@@ -31,9 +31,9 @@
             'documentationUrl' => 'http://www.blitline.com/docs/api',
             'parameters'       => array(
                 'src' => array(
-                    'description' => 'The location of the image you wish to process',
+                    'description' => 'The S3 location of the image you wish to process',
```
Comment (ivan-wolf) [inline src/Detail/Blitline/Client/ServiceDescription/Blitline.php]: Is it an undocumented Blitline feature to pass an array? The documentation (http://www.blitline.com/docs/api) does not mention it.

File: src/Detail/Blitline/Job/Definition/JobDefinition.php [hunk 5]
```diff
@@ -20,16 +20,16 @@ class JobDefinition extends BaseDefinition implements JobDefinitionInterface
     /**
      * @inheritdoc
      */
-    public function setSourceUrl($url)
+    public function setSource(array $src)
```
Comment (ivan-wolf) [inline src/Detail/Blitline/Job/Definition/JobDefinition.php]: Shouldn't we use an object for representation of a source?

**Review by ivan-wolf — CHANGES_REQUESTED** [review]
@curzio-della-santa, please have a look.

File: src/Detail/Blitline/Client/ServiceDescription/Blitline.php (hunk 4 above)
Comment (curzio-della-santa) [reply src/Detail/Blitline/Client/ServiceDescription/Blitline.php]: yes is undocumented, only an example is given.

Blitline Example Jobs (http://docs.blitline.com/articles/examples.html)
- Using the permissions you’ve given Blitline for sources:	By default, Blitline assumes a source is public. If you instead want to have Blitline read OUT of your private bucket, you will need to change how the “src” is described. This is an example of using an “s3” source bucket and key

https://www.blitline.com/v3/home/gist?gist_id=3492389

File: src/Detail/Blitline/Job/Definition/JobDefinition.php (hunk 5 above)
Comment (curzio-della-santa) [reply src/Detail/Blitline/Job/Definition/JobDefinition.php]: creating SourceInterface

**Comment by curzio-della-santa** [comment]
http://blitline.examples.web01.detailnet.me/job-submit.php?image_url=https%3A%2F%2Fs3-eu-west-1.amazonaws.com%2Fcurzio-della-santa%2Ftest-files%2FDesert.jpg

http://blitline.examples.web01.detailnet.me/job-submit-s3.php?image_key=test-files%2FDesert.jpg

processing_with_different_sources

**Pull request merged**